src/Nethermind/Nethermind.Evm/EvmObjectFormat/Handlers/EofV1.cs

```diff
@@ -772,18 +772,21 @@ private static bool ValidateBody(in EofHeader header, ValidationStrategy strateg
     /// </returns>
     private static bool ValidateDataSection(in EofHeader header, ValidationStrategy strategy, ReadOnlySpan<byte> dataBody)
     {
-        // If full body validation is requested, the DataSection size must be less than or equal to the data available.
-        if (strategy.HasFlag(ValidationStrategy.ValidateFullBody) && header.DataSection.Size > dataBody.Length)
+        // If full body validation is requested, or it is runtime mode (but not both)
+        // the DataSection size must be less than or equal to the data available.
+        if ((strategy.HasFlag(ValidationStrategy.ValidateRuntimeMode) ^ strategy.HasFlag(ValidationStrategy.ValidateFullBody))
```

Comment:
Binary xor is a bit wild 😅

Reply:
I've finally legitimately used it, and not to implement some VM opcode.